src/index.js

```diff
@@ -58,22 +60,28 @@ class App extends React.Component {
     });
   }
 
-  handleChangeLoadTranscriptJson(files) {
-    const self = this;
-    const file = files[0];
-    // let type = file.type;
-    // TODO: add checks
-    // let transcriptJsonContent = FileReader.readAsText(file)
-    const fr = new FileReader();
-    fr.onload = function (e) {
-      // e.target.result should contain the text
-      console.log(JSON.parse(e.target.result));
-      self.setState({
-        transcriptData: JSON.parse(e.target.result),
-        // mediaUrl: tedTalkVideoUrl
-      });
-    };
-    fr.readAsText(file);
+  handleChangeLoadTranscriptJson(e) {
+    const files = e.target.files;
+    if (this.state.mediaUrl === null) {
+      e.preventDefault();
+      alert('fist add a video or audio file, and then re-import the transcription');
+    }
+    else {
+      const self = this;
+      const file = files[0];
+      // let type = file.type;
+      // TODO: add checks
+      // let transcriptJsonContent = FileReader.readAsText(file)
+      const fr = new FileReader();
+      fr.onload = function (e) {
```

Comment:
this throws a linting error as `e` is defined in upper scope. rename it (and also would prefer arrow functions if possible)

Reply:
good catch, with the arrow function can also remove the `self = this` at the top.